src/PKSim.Core/Model/PopulationSimulation.cs

```diff
@@ -280,35 +279,43 @@ private DataColumn medianAggregateDataColumns(IReadOnlyList<DataColumn> columns)
          return  new DataColumn(column.Id, column.Dimension, column.BaseGrid)
          {
             Values = Enumerable.Range(0, column.Values.Count).Select(i => 
-                  IndexedValuesFromColumns(columns, i).Median()
+                  indexedValuesFromColumns(columns, i).Median()
                ).ToList()
          };
       }
 
-      private static IReadOnlyList<float> IndexedValuesFromColumns(IReadOnlyList<DataColumn> columns, int i)
+      private static IReadOnlyList<float> indexedValuesFromColumns(IReadOnlyList<DataColumn> columns, int i)
       {
          return columns.Select(x => x.Values[i]).ToList();
       }
 
       public override DataColumn PeripheralVenousBloodColumn(string compoundName)
       {
-         return medianAggregateDataColumns(drugColumnFor(CoreConstants.Organ.PERIPHERAL_VENOUS_BLOOD, CoreConstants.Observer.PLASMA_PERIPHERAL_VENOUS_BLOOD, CoreConstants.Observer.PLASMA_PERIPHERAL_VENOUS_BLOOD, compoundName));
+         return medianAggregateDataColumns(drugColumnsFor(CoreConstants.Organ.PERIPHERAL_VENOUS_BLOOD,
+            CoreConstants.Observer.PLASMA_PERIPHERAL_VENOUS_BLOOD, CoreConstants.Observer.PLASMA_PERIPHERAL_VENOUS_BLOOD,
+            compoundName));
       }
 
       /// <summary>
       ///    tries to find venous blood plasma if defined in the repository. returns null otherwise
       /// </summary>
       public override DataColumn VenousBloodColumn(string compoundName)
       {
-         return medianAggregateDataColumns(drugColumnFor(CoreConstants.Organ.VENOUS_BLOOD, CoreConstants.Compartment.PLASMA, CoreConstants.Observer.CONCENTRATION_IN_CONTAINER, compoundName));
+         return medianAggregateDataColumns(drugColumnsFor(CoreConstants.Organ.VENOUS_BLOOD, 
+            CoreConstants.Compartment.PLASMA, CoreConstants.Observer.CONCENTRATION_IN_CONTAINER, compoundName));
       }
 
       public override DataColumn FabsOral(string compoundName)
       {
-         return medianAggregateDataColumns(drugColumnFor(CoreConstants.Organ.LUMEN, CoreConstants.Observer.FABS_ORAL, CoreConstants.Observer.FABS_ORAL, compoundName));
+         return medianAggregateDataColumns(drugColumnsFor(CoreConstants.Organ.LUMEN, CoreConstants.Observer.FABS_ORAL, CoreConstants.Observer.FABS_ORAL, compoundName));
       }
 
-      private IReadOnlyList<DataColumn> drugColumnFor(string organ, string compartment, string columnName, string compoundName)
+      private DataColumn drugColumnForIndividual(string organ, string compartment, string columnName, string compoundName, int individualId)
+      {
+         return columnsFor(Results.IndividualResultsAsArray().FirstOrDefault(result => result.IndividualId == individualId), organ, compartment, columnName, compoundName);
```

Comment:
There has to be away to access results by Id other than this? This looks very odd to me/.Plus the array conversion etc ..

Reply:
`Results.ResultsFor(individualId)` :) Nice.